woodwork/tests/accessor/test_table_accessor.py

```diff
@@ -2118,26 +2118,34 @@ def test_setitem_invalid_input(sample_df):
         df.ww['signup_date'] = df.signup_date
 
 
-def test_setitem_indexed_column(sample_df):
-    sample_df_with_index = sample_df
-    sample_df_with_index.ww.init()
+def test_setitem_indexed_column_on_indexed_dataframe(sample_df):
```

Comment:
I think this test could use an additional case (you can just put it after the code that's already here) where the index of another column is still present, like:

```python
    sample_df.ww.init(index='id')
    col = sample_df.ww.pop('email')
    col.ww.init(semantic_tags='index')
```

and then after we do `sample_df.ww['id'] = col`, we want to confirm `sample_df.ww.index == 'id'` and that the index tag is no longer in `email`.